pkg/security/rconfig/profiles.go

```diff
@@ -0,0 +1,166 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package rconfig
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	proto "github.com/DataDog/agent-payload/v5/cws/dumpsv1"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	cgroupModel "github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup/model"
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type ProfileConfig struct {
+	Tags    []string
+	Profile []byte
+}
+
+type RCProfileProvider struct {
+	sync.RWMutex
+
+	client *remote.Client
+
+	// tag queue
+	tags    []string
+	nextTag int
+
+	onNewProfileCallback func(selector cgroupModel.WorkloadSelector, profile *proto.SecurityProfile)
+}
+
+// Close stops the client
+func (r *RCProfileProvider) Stop() error {
+	r.client.Close()
+	return nil
+}
+
+func (r *RCProfileProvider) rcProfilesUpdateCallback(configs map[string]state.ConfigCWSProfiles) {
+	log.Infof("new profiles from remote-config policy provider: %v", r.tags)
+
+	// move to the next tag
+	if len(r.tags) > 0 {
+		for _, config := range configs {
+			var profCfg ProfileConfig
+			if err := json.Unmarshal(config.Config, &profCfg); err != nil {
+				log.Errorf("couldn't decode json profile: %s", err)
+				return
```

Comment:
Should we make a `continue` instead of `return` (here and on all subsequent errors). Event if the first profile we get is somehow invalid, we may have valid ones after right?

Reply:
Currently we are getting only one config, but yes you're right I'll change this to a continue